kernel/main.cpp

```diff
@@ -28,6 +28,13 @@
 #include <lib/string.hpp>
 #include <lib/stdio.hpp>
 
+// Used as a magic number for stack smashing protection
```

Comment:
This will only prevent against programming mistakes/errors and is not a good security mechanism. That's fine with me as long as we revisit this once (if) security begins to be a concern.